faker/proxy.py

```diff
@@ -235,3 +243,44 @@ def factories(self):
 
     def items(self):
         return self._factory_map.items()
+
+
+class UniqueProxy:
+    def __init__(self, proxy):
+        self._proxy = proxy
+        self._seen = {}
+        self._sentinel = object()
+
+    def clear(self):
+        self._seen = {}
+
+    def __getattr__(self, name: str):
+        obj = getattr(self._proxy, name)
+        if callable(obj):
+            return self._wrap(name, obj)
+        else:
+            return obj
+
+    def _wrap(self, name, function):
+        @functools.wraps(function)
+        def wrapper(*args, **kwargs):
+            key = (name, *args, tuple(sorted(kwargs.items())))
```

Comment:
Sort the keyword arguments, because `random_int(min=50, max=60)` should be equivalent to `random_int(max=60, min=50)`, although no clue why the signature would be different.